src/configurationGenerator.ts

```diff
@@ -0,0 +1,99 @@
+import * as Globals from './constants'
+import {Shot} from './shot'
+import {BoardState} from './boardState'
+import {Position} from './position'
+import {ShipPlacement} from './shipPlacement'
+
+export class ConfigurationGenerator{
+    public boardCounter: Array<Array<number>>;
+    public squaresHit: Array<Array<boolean>>;
+
+    private nextPath: number[] = [];
+    private exhaustive: boolean = false;
+
+    public constructor(){
+        this.boardCounter = [];
+        this.squaresHit = [];
+        for(var i=0;i <Globals.BOARD_ROWS; i++){
+            this.boardCounter.push([]);
+            this.squaresHit.push([]);
+            for(var j = 0; j<Globals.BOARD_COLS; j++){
+                this.boardCounter[i].push(0);
+                this.squaresHit[i].push(false);
+            }
+        }        
+    }
+
+    public generateConfiguration(lengths:Array<number>, shots: Array<Shot> = [], hitShots:Array<Shot> = []):void{
+        let compatible: boolean = false;
+        let boardToAdd: Array<Array<boolean>>;
+       
+        
+        if(!this.exhaustive){
```

Comment:
Lots of whitespace - if this is intentional (I don't think it is?), I'd say that there's never really any reason to use more than one line of whitespace (two at an absolute maximum in really weird cases).